### PR TITLE
Updates from rspec-dev (2020-11-16)

### PR DESCRIPTION
**Pull request by JonRowe**
These are some updates, generated from rspec-dev's rake tasks.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-on: [pull_request, push]
+on:
+  push:
+    branches:
+      - 'main'
+      - '*-maintenance'
+      - '*-dev'
+  pull_request:
+    branches:
+      - '*'
```
Comment (JonRowe) [inline .github/workflows/ci.yml:13]: @pirj @benoittgt this seems to work for pull requests at least with just one build, will need to push to a branch (probably maintenance to cherry-pick the config) to test it still builds pushes.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,10 +36,6 @@ jobs:
           bundler: ${{ (matrix.ruby == 'jruby-9.1.17.0' && 1) || 2 }}
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
-      - uses: actions/cache@v2
-        with:
-          path: ../bundle
-          key: ${{ runner.os }}-${{ matrix.ruby }}
```
Comment (JonRowe) [inline .github/workflows/ci.yml:34]: This is being removed because it doesn't work, but `bundle-cache: true` does

**Review by pirj — APPROVED** [review]
Great, thanks!

I'll also pull all recent changes to 4-0-dev

**Comment by JonRowe** [comment]
> I'll also pull all recent changes to 4-0-dev

FYI I've rebased all the 4-0-dev branches and set them up so there builds pass.

**Comment by pirj** [comment]
Thanks! :raised_hands:

**Pull request merged**